### PR TITLE
Register the store as a service

### PR DESCRIPTION
**Pull request by martndemus**
I propably want to add a test.
And also people want to feedback on this.

#2818

### CODE REVIEW

File: packages/ember-data/lib/initializers/store.js [hunk 1]
```diff
@@ -33,5 +33,6 @@ export default function initializeStore(container, application) {
 
   // Eagerly generate the store so defaultStore is populated.
   // TODO: Do this in a finisher hook
-  container.lookup('store:main');
+  var store = container.lookup('store:main');
+  container.register('service:store', store,  { instantiate: false });
```
Comment (bmac) [inline packages/ember-data/lib/initializers/store.js]: Travis failed because there are 2 spaces after `store,` which made jscs unhappy.

Comment (martndemus) [inline packages/ember-data/lib/initializers/store.js]: I'll fix it :)

**Comment by martndemus** [comment]
The Ember used in the integration tests is 1.9.1, so I can't add a test to check if the actual injection works. (Though it works on my application with Ember 1.10.0)

**Comment by bmac** [comment]
On travis tests get run against the latest release, beta and canary channels. If you are testing locally you can change the ember version using the `emberchannel` query parameter (e.g. `http://localhost:4200/?hidepassed&emberchannel=beta` or `http://localhost:4200/?hidepassed&emberchannel=canary`).

It should be ok to wrap the test in a check for the existence of `Ember.inject` or `Ember.inject.service` that way the test will only get run in the versions of Ember that support that API. We can remove the check later once `Ember.inject.service` makes it into a release version of Ember.

**Comment by martndemus** [comment]
@bmac Thanks for the pointer!

**Comment by bmac** [comment]
This looks good to me. @martndemus do you mind squashing your commits?

@tomdale would you like to review this pr?

**Comment by martndemus** [comment]
@bmac I squashed the commits.

**Comment by slindberg** [comment]
Yay!

It might also make sense to [extend `DS.Store`](https://github.com/emberjs/data/blob/v1.0.0-beta.15/packages/ember-data/lib/system/store.js#L174) from [`Ember.Service`](https://github.com/emberjs/ember.js/blob/v1.11.0-beta.4/packages/ember-runtime/lib/system/service.js#L38) if it is available.

**Comment by bmac** [comment]
@martndemus I chatted with @tomdale at Ember Conf and he is on board with this change. I also like @slindberg's suggestion. Could you extend the store form `Ember.Service` when available?

File: packages/ember-data/lib/system/store.js [hunk 2]
```diff
@@ -99,6 +99,11 @@ var Store;
 
 var camelize = Ember.String.camelize;
 
+var Service = Ember.Service;
+if (!Service) {
+  Service = Ember.Object;
+}
+
```
Comment (martndemus) [inline packages/ember-data/lib/system/store.js]: This was necessary to be backward compatible with  `ember-data <= 1.9.1`

**Comment by bmac** [comment]
Thanks @martndemus!

**Pull request merged**